src/main/java/org/apache/sling/models/impl/ModelAdapterFactory.java

```diff
@@ -495,12 +480,97 @@ public RuntimeException inject(InjectableElement element, Object value) {
         }
     }
 
+    @NotNull
+    private InjectableElement getElement(InjectableElement element, Type type) {
```

Comment:
IIUC, this method is not named intuitively. It should be something like `getOptionalTypeElement` or something like that.

It also might be best to move this logic to the constructor of `InjectableField` or at least move this anonymous inner class to a new class named `OptionalTypedInjectableField` (or something like that since `Optional` is now a bit ambiguous).

Reply:
Thanks for this idea, will give it a try

Reply:
Moved implementation to a named class `OptionalTypedInjectableElement`